mlflow/server/handlers.py

```diff
@@ -59,6 +61,18 @@ def _get_request_message(request_message, flask_request=request):
     return request_message
 
 
+def catch_mlflow_exception(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except MlflowException as e:
+            response = Response(mimetype='application/json')
+            response.set_data(e.serialize_as_json())
```

Comment:
Because we aren't explicitly setting a status code here, I'm assuming that the response will have status code 200? 

However, 200 is probably not accurate in most cases. While we can handle the 200 code and surface the underlying appropriately in the Python API, requests made directly from REST clients will see successful responses with error text.

In some cases, it makes sense to return a 4xx because the operation signifies a client error (as is the case with `mlflow.start_run(run_uuid=<UUID_OF_DELETED_RUN>)`.

In other cases, we may want to return 5xx (i.e when we can't assign an error code more specific than `mlflow.protos.service_pb2.INTERNAL_ERROR`).

What are your thoughts here?

Reply:
Yeah this is an oversight. Let's change it to always return 500 like before and change it in the future to return error codes based on the `error_code` field.

Reply:
Sounds good!